### DISCUSSION TITLE
Ruby 3 support

### DISCUSSION THREAD

**Discussion by reob-info** [jruby/jruby#6691] [General]
Will JRuby support Ruby 3?
The latest version of Ruby supported is Ruby 2.5.
If support to Ruby 3 will come, there are a timeline to have a final version?
Thanks.

**Comment by enebo** [comment]
We plan on having jruby 9.3 which targets 2.6.x out in a couple of weeks (we are closing out final issues on it now).
We are skipping Ruby 2.7 support and going straight to 3.0.x for our 9.4 release.
The 9.4 release will for sure be out before Rubyconf this fall but we are hopeful we can deliver it months before that time frame.

**Reply by reob-info** [reply]
Nice! Thanks.

**Reply by AlexWayfer** [reply]
> We are skipping Ruby 2.7 support and going straight to 3.0.x for our 9.4 release.
> The 9.4 release will for sure be out before Rubyconf this fall but we are hopeful we can deliver it months before that time frame.

Any updated news/plans?

**Reply by enebo** [reply]
We still plan on 9.4 before Rubyconf next month but it will be going to 3.1.x support and not 3.0.x.  Things are very close.  All language features are complete except expression order evaluation (which will be done in the next week).  Then it is just fixing the remaining red we have in CI which has steadily been dropping.  We decided to not support Ractors initially and will wait to see who wants that support.
